### ISSUE TITLE
`prefer-continue` should allow `return`

### ISSUE THREAD

**Issue by fregante** [sindresorhus/eslint-plugin-unicorn#3408]
Confusing suggestion here:

```ts
async function hideAuthorMetadata(summaryRow: HTMLElement): Promise<void> {
	for (const child of summaryRow.childNodes) {
		if (condition(child)) {
			child.remove();
			return;
		}
	}

	throw new Error('Unable to find the PR metadata text node');
}
```

is auto fixed to:

```ts
async function hideAuthorMetadata(summaryRow: HTMLElement): Promise<void> {
	for (const child of summaryRow.childNodes) {
		if (!condition(child)) {
			continue;
		}

		child.remove();
		return;
	}

	throw new Error('Unable to find the PR metadata text node');
}
```

which seems unnecessary as it adds a statement, keeps the `return`, flips the condition. I think the original version is better here

**Comment by fregante** [comment]
It seems that the rule is fine as long as `return` isn't present
